app/components/PagesList.tsx

```diff
@@ -28,103 +34,59 @@ export interface PagesListProps {
-            start: (event, ui) => {
-                pageIndex = ui.item.index();
-            },
-            // @ts-ignore until the application uses jQuery
-            update: (event, ui) => onPageReorder(pageIndex, ui.item.index())
-        });
-        this.enableReorderInEditMode();
-    }
 
-    componentDidUpdate(/* prevProps, prevState */) {
-        this.enableReorderInEditMode();
-    }
+export default function PagesList({
+    isEditMode,
+    onPageRemoved,
+    onPageReorder,
+    onPageSelected,
+    pages,
+    selected = ''
+}: PagesListProps) {
+    const [pageToRemove, setPageToRemove] = useState<Page | null>(null);
+    const pageIds = _.filter(pages, p => !p.isDrillDown).map(({ id }) => id);
+    const sensors = useSensors(useSensor(PointerSensor));
 
-    enableReorderInEditMode() {
-        const { isEditMode } = this.props;
-        if (isEditMode) {
-            // @ts-ignore until the application uses jQuery
-            if ($(this.pagesRef.current).sortable('option', 'disabled')) {
-                // @ts-ignore until the application uses jQuery
-                $(this.pagesRef.current).sortable('enable');
-            }
-            // @ts-ignore until the application uses jQuery
-        } else if (!$(this.pagesRef.current).sortable('option', 'disabled')) {
-            // @ts-ignore until the application uses jQuery
-            $(this.pagesRef.current).sortable('disable');
+    let pageCount = 0;
+    pages.forEach(page => {
+        if (!page.isDrillDown) {
+            pageCount += 1;
         }
-    }
+    });
+    const handleDragEnd = useCallback(
+        (event: DragEndEvent) => {
+            const { active, over } = event;
+            if (active && over && active.id !== over.id) {
+                const oldIndex = pageIds.indexOf(active.id);
+                const newIndex = pageIds.indexOf(over.id);
 
-    render() {
-        const { isEditMode, onPageRemoved, onPageSelected, pages, selected } = this.props;
-        const { pageToRemove } = this.state;
-        let pageCount = 0;
-        pages.forEach(page => {
-            if (!page.isDrillDown) {
-                pageCount += 1;
+                onPageReorder(oldIndex, newIndex);
             }
-        });
+        },
+        [pageIds]
+    );
 
-        return (
-            <>
-                <div className="pages" ref={this.pagesRef}>
-                    {_.filter(pages, p => !p.isDrillDown).map(
-                        page => (
-                            <Menu.Item
+    return (
+        <>
+            <DndContext
+                sensors={sensors}
+                collisionDetection={closestCenter}
+                onDragEnd={isEditMode ? handleDragEnd : undefined}
+            >
+                <SortableContext items={isEditMode ? pageIds : []} strategy={verticalListSortingStrategy}>
+                    <div className="pages">
+                        {_.filter(pages, p => !p.isDrillDown).map(page => (
```

Comment:
As we discussed, here - if you are sure that `pages` are defined - you can probably use:
```suggestion
                        {pages.filter(page => !page.isDrillDown).map(page => (
```